autoscaler/controllers/datacollection/configmap.go

```diff
@@ -283,3 +285,42 @@ func getConfigMap(ctx context.Context, c client.Client, namespace string) (*v1.C
 
 	return configMap, nil
 }
+
+func getSignalsFromOtelcolConfig(otelcolConfigContent string) ([]common.ObservabilitySignal, error) {
+	config := config.Config{}
+	err := yaml.Unmarshal([]byte(otelcolConfigContent), &config)
```

Comment:
This unmarshilng can have bad impact on memory consumption and CPU if the reconciler is out of control.

Reply:
This function should be called at most once every 10 seconds https://github.com/odigos-io/odigos/blob/0f80c84faefd58373912e50f402b0d5e3712076e/autoscaler/controllers/datacollection/root.go#L69

To be honest, I would love for the collector to automatically reload configuration in the future and then we can get rid of this code altogether